### ISSUE TITLE
include paths too long

### ISSUE THREAD

**Issue by tkegrund** [SerialKeyManager/SKM-Client-API-CPP#14]
while building cryptolense in one of our applications, we hit the maximal path length "path too long" on windows.
Some includes are unnecessary long and are causing this problem:

example: cryptolens\src\basic_SKM.cpp
154 characters:
cryptolens\include\cryptolens/imports/ArduinoJson5/../../../../third_party/ArduinoJson5/ArduinoJson/StringTraits/../TypeTraits/../Polyfills/alias_cast.hpp

**Comment by svedi** [comment]
In order for us to reproduce this problem, would it be possible for you to provide some information about your environment? Like which version of Windows you are using and which compiler?

**Comment by tkegrund** [comment]
Microsoft Windows [Version 10.0.17763.3165]
clang version 13.0.1

**Comment by svedi** [comment]
We pushed a change that hopefully fixes this to the preview branch. The idea is that when the preprocessor definition `CRYPTOLENS_SHORT_INCLUDE_PATHS` is set we just include the needed files directly without going through the relative path. However, then additional paths need to be added to the include path.

I added an example in `examples/unix/cmake/CMakeLists.txt` which shows how setting the preprocessor definition and adding the include paths can be done in CMake. Does this should like a solution which would work for you?
